.github/workflows/ci_cmake.yml

```diff
@@ -0,0 +1,130 @@
+          fi
+
+  macos-native:
+    name: 'test macos'
+    runs-on: macos-14
+    env:
+      BASE_ROOT_DIR: ${{ github.workspace }}
+      CCACHE_DIR: ${{ github.workspace }}/ccache
+      BASE_BUILD_DIR: ${{ github.workspace }}/build
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Clang version
+        run: |
+          if [ -d "/Applications/Xcode_15.0.app" ]; then
+            sudo xcode-select --switch /Applications/Xcode_15.0.app
+          else
+            echo "Xcode 15.0 not found, skipping xcode-select"
+          fi
+          clang --version
+
+      - name: Setup Homebrew packages
+        env:
+          HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK: 1
+        run: |
+          brew install --quiet python@3 || brew link --overwrite python@3
+          brew install --quiet coreutils ninja pkgconf gnu-getopt ccache boost libevent zeromq qt@5 qrencode berkeley-db4
+
+      - name: Restore Ccache cache
+        id: ccache-cache
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ github.job }}-ccache-${{ github.run_id }}
+          restore-keys: ${{ github.job }}-ccache-
+
+      - name: Enable core dumps
+        run: ulimit -c unlimited
+
+      - name: Compile and run tests
+        run: |
+          BITCOIN_CONFIG_ALL="$BITCOIN_CONFIG_ALL -DCMAKE_INSTALL_PREFIX=$BASE_OUTDIR"
+          cmake -S "$BASE_ROOT_DIR" -B "$BASE_BUILD_DIR" ${BITCOIN_CONFIG_ALL} ${BITCOIN_CONFIG}
+          cmake --build build --parallel -j $(nproc) --target all
+          export LD_LIBRARY_PATH="${DEPENDS_DIR}/${HOST}/lib"
+          export CTEST_OUTPUT_ON_FAILURE=ON
+          echo "---------------- Build unit test ---------------- "
+          cmake --build build --target test_tapyrus -j $(nproc)
+          echo "---------------- Start unit test ---------------- "
+          cd ${{ github.workspace }}/build/src/test
+          ctest --output-on-failure -j 1 --timeout 300
+          echo "----------------  Start util test ---------------- "
+          cd ${{ github.workspace }}
+          cmake --build build --target check-tapyrus-util -j $(nproc)
+          echo "----------------  Start functional test ---------------- "
+          cd ${{ github.workspace }}
+          ${{ github.workspace }}/test/functional/test_runner.py -j $(( $(nproc) * 2 )) --combinedlogslen=99999999 --coverage --failfast --quiet 
+          echo "---------------- DONE ---------------------"
+
+      - name: Debug Core Dump (if exists)
+        if: failure()
+        run: |
+          if [ -f core ]; then
+            gdb -batch -ex "bt" -ex "quit" ./test_binary core
+          fi
+
+      - name: Save Ccache cache
+        uses: actions/cache/save@v4
+        if: steps.ccache-cache.outputs.cache-hit != 'true'
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ github.job }}-ccache-${{ github.run_id }}
```

Comment:
What about existing WIN64 tests?

Reply:
I haven't attempted the cross compilation for windows using make yet. Let me fix linux and macOS with the new sdk first. 
This would still run using the gnu autotools ci.